cmake/ProhibitInSourceBuild.cmake

```diff
@@ -0,0 +1,23 @@
+
+# Distributed under the University of Illinois/NCSA Open Source License.
+# See LICENSE.txt for details.
+
+function(prohibit_in_source_build)
```

Comment:
Why don't we not have this be a function so that just including the file activates the requirement? I think that would make more sense, then we have less code to deal with.